p2p/host/autorelay/autorelay_test.go

```diff
@@ -224,6 +224,7 @@ func TestBackoff(t *testing.T) {
 	defer h.Close()
 
 	require.Eventually(t, func() bool {
+		cl.Add(1)
```

Comment:
We shouldn't do this. It's at best a hack. Instead #2187 should fix this.

Reply:
I'm fine with rebasing these changes on that branch. 

Another thing we can do would be to expose a channel which is closed when the background go routine has initialised. This cl.Add is required because we need to increment the timer once after creating a timer for it to trigger.

For these changes we can also fix this in the library itself. We should fire the timer when the timer is created.